application/src/test/java/org/togetherjava/tjbot/AbstractJDMock.java

```diff
@@ -0,0 +1,96 @@
+package org.togetherjava.tjbot;
+
+import net.dv8tion.jda.api.entities.SelfUser;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.JDAImpl;
+import net.dv8tion.jda.internal.entities.EntityBuilder;
+import net.dv8tion.jda.internal.entities.PrivateChannelImpl;
+import net.dv8tion.jda.internal.entities.SelfUserImpl;
+import net.dv8tion.jda.internal.entities.UserImpl;
+import net.dv8tion.jda.internal.interactions.CommandInteractionImpl;
+import net.dv8tion.jda.internal.requests.Requester;
+import net.dv8tion.jda.internal.requests.restaction.MessageActionImpl;
+import net.dv8tion.jda.internal.utils.config.AuthorizationConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class mocks a valid JDA instance with a user, an api and a message channel mock
+ * implementations
+ */
+public abstract class AbstractJDMock {
```

Comment:
name should likely have been `JDA` not just `JD`. That said, please stick to PascalCase. So `AbstractJdaMock`